binfmt/binfmt_execmodule.c

```diff
@@ -164,23 +164,23 @@ int exec_module(FAR const struct binary_s *binp,
 #if defined(CONFIG_ARCH_ADDRENV) && defined(CONFIG_BUILD_KERNEL)
   /* Instantiate the address environment containing the user heap */
 
-  ret = up_addrenv_select(&binp->addrenv, &oldenv);
+  ret = addrenv_select((struct addrenv_s *)&binp->addrenv);
```

Comment:
why need cast(`(struct addrenv_s *)`) here?

Reply:
Because binp is declared as const here, but the reference counter is needed in addrenv_select() so it must be made mutable

Reply:
should we change binp prototype to reflect the change?

Reply:
Sure, I can do that